app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -219,23 +202,33 @@ class BrowserFragment : IWebViewLifecycleFragment() {
 
         context?.getAccessibilityManager()?.removeTouchExplorationStateChangeListener(touchExplorationStateChangeListener)
         touchExplorationStateChangeListener = null
+    }
 
-        // Since we start the async jobs in View.init and Android is inflating the view for us,
-        // there's no good way to pass in the uiLifecycleJob. We could consider other solutions
-        // but it'll add complexity that I don't think is probably worth it.
-        homeScreen.uiLifecycleCancelJob.cancel(CancellationException("Parent lifecycle has ended"))
+    private fun setNavigationOverlayIsVisible(isVisible: Boolean, isHomescreenOnStartup: Boolean = false) {
+        callbacks?.onHomeVisibilityChange(isVisible, isHomescreenOnStartup = isHomescreenOnStartup)
+        updateWebViewVisibility(isVoiceViewEnabled = context!!.isVoiceViewEnabled(), isHomescreenOnStartup = isHomescreenOnStartup)
+
+        if (isVisible) {
+            NavigationOverlayFragment.newInstance(isInitialHomescreen = isHomescreenOnStartup)
+                .show(fragmentManager!!, activity!!)
```

Comment:
They load in fast enough for the modest number of home tiles I tested but the animation performance suffers due to the dynamic loading. I filed #160 to cache the images, as opposed to the fragment, in memory, which I think is 1) a more generalized solution that 2) decouples the image caching from the fragment caching, and 3) is a better user experience because we can simply drop the images from memory if the user is resource constrained (e.g. with Picasso), unlike writing the complex code to dynamically allocate and deallocate a fragment.